pkg/scheduler/framework/plugins/dynamicresources/structuredparameters_test.go

```diff
@@ -0,0 +1,1259 @@
+						Allocation: &resourceapi.AllocationResult{
+							ResourceHandles: []resourceapi.ResourceHandle{
+								{
+									DriverName: "driver-a",
+									StructuredData: &resourceapi.StructuredResourceHandle{
+										NodeName: "node-b",
+										Results: []resourceapi.DriverAllocationResult{{
+											AllocationResultModel: resourceapi.AllocationResultModel{
+												NamedResources: &resourceapi.NamedResourcesAllocationResult{
+													Name: "X",
+												},
+											},
+										}},
+									},
+								},
+								{
+									DriverName: "driver-b",
+									StructuredData: &resourceapi.StructuredResourceHandle{
+										NodeName: "node-b",
+										Results: []resourceapi.DriverAllocationResult{{
+											AllocationResultModel: resourceapi.AllocationResultModel{
+												NamedResources: &resourceapi.NamedResourcesAllocationResult{
+													Name: "X",
+												},
+											},
+										}},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			expectResources: resources{
+				"node-b": map[string]ResourceModels{},
+			},
+		},
+
+		"in-flight": {
+			slices: sliceList{
+				&resourceapi.ResourceSlice{
+					NodeName:   "node",
+					DriverName: "driver",
+					ResourceModel: resourceapi.ResourceModel{
+						NamedResources: &resourceapi.NamedResourcesResources{
+							Instances: []resourceapi.NamedResourcesInstance{{Name: "one"}},
+						},
+					},
+				},
+			},
+
+			claims: claimList{
+				&resourceapi.ResourceClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						UID: "abc",
+					},
+					// Allocation not recorded yet.
+				},
+			},
+
+			inFlight: map[types.UID]resourceapi.ResourceClaimStatus{
+				"abc": {
+					DriverName: "driver",
+					Allocation: &resourceapi.AllocationResult{
+						ResourceHandles: []resourceapi.ResourceHandle{{
+							DriverName: "driver",
+							StructuredData: &resourceapi.StructuredResourceHandle{
+								NodeName: "node",
+								Results: []resourceapi.DriverAllocationResult{{
+									AllocationResultModel: resourceapi.AllocationResultModel{
+										NamedResources: &resourceapi.NamedResourcesAllocationResult{
+											Name: "one",
+										},
+									},
+								}},
+							},
+						}},
+					},
+				},
+			},
+
+			expectResources: resources{"node": map[string]ResourceModels{
+				"driver": {
+					NamedResources: namedresourcesmodel.Model{
+						Instances: []namedresourcesmodel.InstanceAllocation{
+							{
+								Allocated: true,
+								Instance:  &resourceapi.NamedResourcesInstance{Name: "one"},
+							},
+						},
+					},
+				},
+			}},
+		},
+	}
+
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			tCtx := ktesting.Init(t)
+
+			var inFlightAllocations sync.Map
+			for uid, claimStatus := range tc.inFlight {
+				inFlightAllocations.Store(uid, &resourceapi.ResourceClaim{Status: claimStatus})
+			}
+
+			slices := tc.slices
+			if slices == nil {
+				slices = sliceList{}
+			}
```

Comment:
`slices` is an interface because some tests use `sliceList` and others use `sliceError`. The interface value must not be nil, because then there is no implementation.